File: calculator.cpp
#include <iostream>

using string = std::string;

const string EXIT = "exit";

string[] tokenized_line(string line){
  string tokens[] = [];
}

bool should_exit(string line){
  return line == EXIT;
}

int main() {
  while(true) {
    string line;
    std::cout<< "> ";
    std::cin >> line;

    if(should_exit(line)){
      break;
    }

    std::cout << "Evaluates to: "<< line << std::endl;
  }

  std::cout << "Goodbye :wave:" << std::endl;
  return 0;
}

File: calculator.py
import tokenizer
from evaluator import Evaluator

EXIT = "exit"

# Evaluate input
def evaluate_input(line):
    if line == EXIT:
        quit(1)
    tokens = tokenizer.tokenize_line(line)
    evaluator = Evaluator(tokens)
    result = evaluator.expr()
    print(result)

# Take in a line of input from std_in
def get_input():
    return input("calc> ")

# Take in input and evalutate the input
def driver():
    while True:
        line = get_input()
        evaluation = evaluate_input(line)

driver()
